extension/intents/navigation/navigation.js

```diff
@@ -32,23 +32,30 @@ intentRunner.registerIntent({
   name: "navigation.navigate",
   async run(context) {
     const query = context.slots.query;
-    const where = context.slots.where;
-    const cached = queryDatabase.get(query.toLowerCase());
-    if (where === "window") {
-      if (cached) {
-        await browser.windows.create({ url: cached.url });
+    const result = await browser.storage.sync.get("pageNames");
+    const pageNames = result.pageNames;
+    const savedUrl = pageNames[query];
+    if (savedUrl) {
+      await context.openOrFocusTab(savedUrl);
+    } else {
+      const where = context.slots.where;
+      const cached = queryDatabase.get(query.toLowerCase());
+      if (where === "window") {
```

Comment:
Right now this can't be invoked for a named tab because of phrasing constraints ("open name page intent in new window"), but the logic would want to be switched around once that was possible.

Reply:
This works for nicknames - words or phrases - that do not closely match "matches" (I checked it with different nicknames). If I understand you correctly, this will need to be changed once the way page names are registered is changed